backend/director/agents/text_to_movie.py

```diff
@@ -0,0 +1,509 @@
+
+        self.engine_configs = {
+            "kling": EngineConfig(
+                name="kling",
+                max_duration=10,
+                preferred_style="cinematic",
+                prompt_format="detailed",
+            ),
+            "stabilityai": EngineConfig(
+                name="stabilityai",
+                max_duration=4,
+                preferred_style="photorealistic",
+                prompt_format="concise",
+            ),
+        }
+        super().__init__(session=session, **kwargs)
+
+    def run(
+        self,
+        collection_id: str,
+        engine: str = "stabilityai",
+        job_type: str = "text_to_movie",
+        text_to_movie: Optional[dict] = None,
+        *args,
+        **kwargs,
+    ) -> AgentResponse:
+        """
+        Process the storyline to generate a movie.
+
+        :param collection_id: The collection ID to store generated assets
+        :param engine: Video generation engine to use
+        :return: AgentResponse containing information about generated movie
+        """
+        try:
+            self.videodb_tool = VideoDBTool(collection_id=collection_id)
+            self.output_message.actions.append("Processing input...")
+            video_content = VideoContent(
+                agent_name=self.agent_name,
+                status=MsgStatus.progress,
+                status_message="Generating movie...",
+            )
+            self.output_message.content.append(video_content)
+            self.output_message.push_update()
+
+            if engine not in self.engine_configs:
+                raise ValueError(f"Unsupported engine: {engine}")
+
+            if engine == "stabilityai":
+                STABILITY_API_KEY = os.getenv("STABILITYAI_API_KEY")
+                if not STABILITY_API_KEY:
+                    raise Exception("Stability AI API key not found")
+                self.video_gen_tool = StabilityAITool(api_key=STABILITY_API_KEY)
+                self.video_gen_config_key = "video_stabilityai_config"
+            elif engine == "kling":
+                KLING_API_ACCESS_KEY = os.getenv("KLING_AI_ACCESS_API_KEY")
+                KLING_API_SECRET_KEY = os.getenv("KLING_AI_SECRET_API_KEY")
+                if not KLING_API_ACCESS_KEY or not KLING_API_SECRET_KEY:
+                    raise Exception("Kling AI API key not found")
+                self.video_gen_tool = KlingAITool(
+                    access_key=KLING_API_ACCESS_KEY, secret_key=KLING_API_SECRET_KEY
+                )
+                self.video_gen_config_key = "video_kling_config"
+            else:
+                raise Exception(f"{engine} not supported")
+
+            # Initialize tools
+            self.audio_gen_tool = ElevenLabsTool(
+                api_key=os.getenv("ELEVENLABS_API_KEY")
+            )
+            self.audio_gen_config_key = "audio_elevenlabs_config"
+
+            # Initialize steps
+            if job_type == "text_to_movie":
```

Comment:
there can be, but it can be either job_type or a config option 

another workflow of text to movie
where a single movie just gets extended

the reason why i won't prefer config here is because this will be a different workflow altogether, it better to keep that under seperate conditions

Reply:
Cool